src/app/ReadClient.cpp

```diff
@@ -150,9 +150,13 @@ CHIP_ERROR ReadClient::SendReadRequest(NodeId aNodeId, Transport::AdminId aAdmin
                     err = CHIP_ERROR_INVALID_ARGUMENT;
                     ExitNow();
                 }
+                attributePathBuilder.EndOfAttributePath();
```

Comment:
Should we consider having the AttributeP|athList::Builder and similar objects auto-close when going out of scope (or alternatively error-log if not closed when running out of scope? Wondering if this kind of failure could have been caught at compile time.